sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -122,6 +122,14 @@ class SparkSession private(
   val sqlContext: SQLContext = new SQLContext(this)
 
   /**
+   * This is the interface through which the user can get [[JavaSparkContext]]
+   *
+   * @since 2.0.0
+   */
+  @transient
+  val javaSparkContext: JavaSparkContext = new JavaSparkContext(sparkContext)
```

Comment:
But when user write code in java and he/she wants the JavaSparkConext, then need to get the sparkContext and create the java context with JavaSparkContext method fromSparkContext(sc) so its more good to provide directly JavaSparkContext from spark session to the user.